index.js

```diff
@@ -162,6 +162,193 @@ Denque.prototype.pop = function pop() {
+};
+
+/**
+ * Remove number of items from the specified index from the list.
+ * Returns array of removed items.
+ * Returns undefined if the list is empty.
+ * @param index
+ * @param count
+ * @returns {array}
+ */
+Denque.prototype.remove = function remove(index, count) {
+  var i = index;
+  var removed;
+  var del_count = count;
+  // expect a number or return undefined
+  if ((i !== (i | 0))) {
+    return void 0;
+  }
+  if (this._head === this._tail) return void 0;
+  var size = this.size();
+  var len = this._list.length;
+  if (i >= size || i < -size || count<1) return void 0;
+  if (i < 0) i += size;
+  if (count === 1 || !count){
+    removed = new Array(1);
+    removed[0] = this.removeOne(i);
+    return removed;
+  }
+  if (i === 0 && i + count >= size) return this.clear();
+  if (i + count > size) count = size - i;
+  var k;
+  removed = new Array(count);
+  for(k = 0; k < count; k++){
+    removed[k] = this._list[(this._head + i + k) & this._capacityMask];
+  }
+  i = (this._head + i) & this._capacityMask;
+  if (index + count === size){
+    this._tail = (this._tail - count + len) & this._capacityMask;
+    for(k = count; k > 0; k--){
+      this._list[i = (i + 1 + len) & this._capacityMask] = void 0;
+    }
+    return removed;
+  }
+  if(index === 0){
+    this._head = (this._head + count + len) & this._capacityMask;
+    for(k = count - 1; k > 0; k--){
+      this._list[i = (i + 1 + len) & this._capacityMask] = void 0;
+    }
+    return removed;
+  }
+  if(index < size / 2){
+    this._head = (this._head + index + count + len) & this._capacityMask;
+    for(k = index; k > 0; k--){
+      this.unshift(this._list[i = (i - 1 + len) & this._capacityMask]);
+    }
+    i = (this._head - 1 + len) & this._capacityMask;
+    while(del_count>0){
+      this._list[i = (i - 1 + len) & this._capacityMask] = void 0;
+      del_count--;
+    }
+  }else{
+    this._tail = i;
+    i = (i + count + len) & this._capacityMask;
+    for(k = size - (count + index); k > 0; k--){
+      this.push(this._list[i++]);
+    }
+    i = this._tail;
+    while(del_count>0){
+      this._list[i = (i + 1 + len) & this._capacityMask] = void 0;
+      del_count--;
+    }
+  }
+  if (this._head < 2 && this._tail > 10000 && this._tail <= len >>> 2) this._shrinkArray();
+  return removed;
+};
+
+/**
+ * Native splice implementation.
+ * Remove number of items from the specified index from the list and/or add new elements.
+ * Returns array of removed items or empty array if count == 0.
+ * Returns undefined if the list is empty.
+ * 
+ * @param index
+ * @param count
+ * @param {...*} [elements]
+ * @returns {array}
+ */
+Denque.prototype.splice = function splice(index, count) {
+  var i = index;
+  var size = this.size();
+  // expect a number or return undefined
+  if ((i !== (i | 0))) {
+    return void 0;
+  }
+  if (this._head === this._tail) return void 0;
+  if (i > size || i < -size) return void 0;
+  if (i === size && count != 0) return void 0;
```

Comment:
Not entirely sure what this line is trying to do? Could you clarify - your PR is missing quite a bit of coverage so i'm trying to bring it back up to 99%~